orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java

```diff
@@ -377,7 +399,11 @@ public void initServiceRun(List<GatewayConfig> allGateway, Set<Node> allNodes, S
             }
             grainUploader.uploadGrains(allNodes, saltConfig.getGrainsProperties(), exitModel, sc, exitCriteria);
 
+            // TODO Question: Is SyncAllRunner needed? This takes about 1.8s. Operations seem to work fine without this.
```

Comment:
This help us to make sure the newly added pillars are properly used for the next Highstate.

Reply:
we had some intermittent, mysterious issues before adding it

Reply:
ok. in the grand scheme of adding multiple nodes - this adds time but there's much bigger items.